crates/block-producer/src/store/mod.rs

```diff
@@ -121,17 +121,19 @@ impl TryFrom<GetTransactionInputsResponse> for TransactionInputs {
 // STORE CLIENT
 // ================================================================================================
 
+type InnerClient = store_client::ApiClient<InterceptedService<Channel, OtelInterceptor>>;
+
 /// Interface to the store's gRPC API.
 ///
 /// Essentially just a thin wrapper around the generated gRPC client which improves type safety.
 #[derive(Clone)]
 pub struct StoreClient {
-    inner: store_client::ApiClient<Channel>,
+    inner: InnerClient,
 }
 
 impl StoreClient {
     /// TODO: this should probably take store connection string and create a connection internally
-    pub fn new(store: store_client::ApiClient<Channel>) -> Self {
+    pub fn new(store: InnerClient) -> Self {
```

Comment:
Is the `TODO` still relevant here?

Reply:
I think so. Or at least I think the api/usage could be improved. It feels a bit clunky to use and maybe needs a revisit at some stage, though its relatively low priority.

I imagine this would be annoying to use in e.g. test setups - where we don't care about otel etc.